core/src/plugins/filed/BareosFdPluginLibcloud.py

```diff
@@ -0,0 +1,386 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# This program is Free Software; you can redistribute it and/or
+# modify it under the terms of version three of the GNU Affero General Public
+# License as published by the Free Software Foundation, which is
+# listed in the file LICENSE.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+#
+# Author: Alexandre Bruyelles <git@jack.fr.eu.org>
+#
+
+import BareosFdPluginBaseclass
+import bareosfd
+import bareos_fd_consts
+import ConfigParser as configparser
+import datetime
+import dateutil.parser
+from bareos_libcloud_api.bucket_explorer import JOB_TYPE
+from bareos_libcloud_api.debug import debugmessage
+from bareos_libcloud_api.debug import jobmessage
+from bareos_libcloud_api.debug import set_plugin_context
+import io
+import itertools
+import libcloud
+import multiprocessing
+import os
+from time import sleep
+
+from BareosLibcloudApi import SUCCESS
+from BareosLibcloudApi import ERROR
+from BareosLibcloudApi import BareosLibcloudApi
+from bareos_fd_consts import bRCs, bCFs, bIOPS, bJobMessageType, bFileType
+from libcloud.storage.types import Provider
+from libcloud.storage.types import ObjectDoesNotExistError
+from sys import version_info
+
+
+class FilenameConverter:
+    __pathprefix = "PYLIBCLOUD:/"
+
+    @staticmethod
+    def BucketToBackup(filename):
+        return FilenameConverter.__pathprefix + filename
+
+    @staticmethod
+    def BackupToBucket(filename):
+        return filename.replace(FilenameConverter.__pathprefix, "")
+
+
+def str2bool(data):
```

Comment:
Why not use distutils.util.strtobool(val)?

Reply:
The intent was to simply ignore the case of True or true. Will fix this anyway.